src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -174,13 +164,13 @@ def get_dependencies(self):
             )
 
         deps = [
-            self.base._make_install_req(str(r))
+            factory.make_requirement_from_spec(str(r), self.base._ireq)
```

Comment:
nit: I'm a bit sad that `ExtrasCandidate` now relies on `_ireq` which is an implementation detail of `base`, but I don't see a cleaner way of doing this.

Reply:
Same 😥